src/Filesystem.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Yiisoft\Files;
+
+use League\Flysystem\Filesystem as LeagueFilesystem;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\PathNormalizer;
+use Yiisoft\Aliases\Aliases;
+
+class Filesystem extends LeagueFilesystem implements FilesystemInterface
+{
+    private Aliases $aliases;
+
+    public function __construct(FilesystemAdapter $adapter, array $aliases = [], array $config = [], PathNormalizer $pathNormalizer = null)
+    {
+        if ($aliases !== []) {
+            $aliases = array_merge(['@root' => ''], $aliases);
+            $aliases['@root'] = '';
```

Comment:
Something here is superfluous. Either array_merge or assignment.

Reply:
You're wrong, @root can be not empty after merge.

Reply:
I See, Thank You